### PR TITLE
Set 1000 only if 100 < disk size < 1000 when querying IOPS

### PR DESCRIPTION
**Pull request by mattcui**
For quickly addressing the issue like -> http://stackoverflow.com/questions/36395860/no-data-for-iops-while-selecting-storage-size-250gb-or-500gb-in-performance-stor

### CODE REVIEW

**Comment by mattcui** [comment]
@maximilien I added UTs, please help review again, thanks.

File: services/fakes/fake_softlayer_product_package.go [hunk 1]
```diff
@@ -24,7 +26,24 @@ func (fps *FakeProductPackageService) GetItemsByType(packageType string) ([]data
...
 
 func (fps *FakeProductPackageService) GetItemPrices(packageId int, filters string) ([]datatypes.SoftLayer_Product_Item_Price, error) {
-	return []datatypes.SoftLayer_Product_Item_Price{}, errors.New("Not supported")
+	var textFixtrue = ""
+
+	switch {
+	case strings.Contains(filters, "keyName"):
+		textFixtrue = "SoftLayer_Product_Package_getItemPrices.json"
+	case strings.Contains(filters, "100") || strings.Contains(filters, "1000") || strings.Contains(filters, "2000"):
+		textFixtrue = "SoftLayer_Product_Package_getItemPricesBySizeAndIops.json"
```
Comment (maximilien) [inline services/fakes/fake_softlayer_product_package.go]: What is this `textFixtrue = "SoftLayer_Product_Package_getItemPricesBySizeAndIops.json"` in the middle of the code?

File: services/fakes/fake_softlayer_product_package.go [hunk 2]
```diff
@@ -24,7 +26,24 @@ func (fps *FakeProductPackageService) GetItemsByType(packageType string) ([]data
...
+	}
+
+	if textFixtrue == "" {
+		return []datatypes.SoftLayer_Product_Item_Price{}, errors.New(fmt.Sprintf("No matched IOPS found for filters %s", filters))
+	}
+
+	response, _ := testhelpers.ReadJsonTestFixtures("services", textFixtrue)
```
Comment (maximilien) [inline services/fakes/fake_softlayer_product_package.go]: Using fixtures should be only done in tests not code.

**Comment by maximilien** [comment]
@mattcui this is completely wrong. Let's find some time (maybe 2 hours) on Sunday night or Monday to pair on this. I think you are completely missing the point.

Do you have a [screenhero](https://screenhero.com/account/) account? If not then let me know and I'll send you an invite.

**Review by maximilien — CHANGES_REQUESTED** [review]
This is wrong. See my comment below. You are mixing code and tests together.

File: services/fakes/fake_softlayer_product_package.go (hunk 2 above)
Comment (mattcui) [inline services/fakes/fake_softlayer_product_package.go]: I referenced [this function](https://github.com/maximilien/softlayer-go/blob/master/services/fakes/fake_softlayer_product_package.go#L17-L24) to write my code. I think I could use another way to fake functions in [softlayer_product_package.go](https://github.com/maximilien/softlayer-go/blob/master/services/softlayer_product_package.go)

**Comment by mattcui** [comment]
@maximilien I rewrote this part of UT by leveraging [counterfeiter](https://github.com/maxbrunsfeld/counterfeiter) UT framework. Please help review. If you agree with this approach, I will open another bug or chore to fix the wrong UT code in [fake_softlayer_product_package.go](https://github.com/maximilien/softlayer-go/blob/master/services/fakes/fake_softlayer_product_package.go#L17-L24). Please let me know if it's not what you want yet, we could book a time to pair on this. Thanks.

**Review by maximilien — APPROVED** [review]
OK this is much better. I am merging for expediency, but you need to do two follow ups:
1. make sure this works for existing clients using these kind of disks. So deliver beta release and make sure results are the same
2. I would like to understand what disk sizes we will be missing with the new `selectIopsItemPriceIdOnSizebyLevel` which I feel could have an `"exact"` option when user wants to specify the exact disk size or nothing...

**Pull request merged**